phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -2639,6 +2646,20 @@ private MetaDataMutationResult doDropTable(byte[] key, byte[] tenantId, byte[] s
                         QueryServices.ALLOW_SPLITTABLE_SYSTEM_CATALOG_ROLLBACK);
                 }
             }
+            if (settingNewPhoenixTTLAttribute(tableMetadata, PHOENIX_TTL_BYTES)) {
+                // Disallow if the parent has PHOENIX_TTL set.
+                if (parentTable != null &&  parentTable.getPhoenixTTL() != PHOENIX_TTL_NOT_DEFINED) {
+                    isSchemaMutationAllowed = false;
+                }
+
+                // Since we do not allow propagation of PHOENIX_TTL values during ALTER for now.
+                // If a child view exists and the parent previously had a PHOENIX_TTL value set then that
+                // implies that the child view too has a valid PHOENIX_TTL (non zero).
+                // In this case we do not allow for ALTER of the parent view PHOENIX_TTL value.
+                if (!childViews.isEmpty()) {
```

Comment:
What if we're setting the TTL for the first time for a parent view i.e. it was PHOENIX_TTL_NOT_DEFINED first and now we're setting it for the first time ever. This check will disallow that too. Is that intended?

Reply:
Will change the comment to be less confusing.